### PR TITLE
dragon's breath and HE shells

### PR DESCRIPTION
**Pull request by SonixApache**
Adds Dragon's Breath and Explosive shotgun shells.
The DB ones fire a plume of fire, much like a flamethrower. They can be printed from a hacked ammolathe.
The Explosive/Fragmentation ones cause a (0,0,1,5) explosion on the point of impact. They're unobtainable for now.
Also added a unique sprite for buckshot shell boxes because I forgot last time.

boxes_and_shells
FIRE
BOOM

🆑 
 - rscadd: Added 12 gauge Dragon's Breath and Explosive shells.
 - imageadd: Buckshot shell boxes now have their own sprite.

### CODE REVIEW

**Comment by DrSnips** [comment]
Post the local atmos stats after firing one.

**Comment by SonixApache** [comment]
image
uses modern flamethrower plume logic without oxygen so it doesn't heat anything up

**Comment by DrSnips** [comment]
That's a huge relief.

**Comment by Exxion** [comment]
Did somebody just discover the reaction button?
Anyway, for the record, I'm pretty sure the fourth explosion number doesn't do anything

**Comment by Caboozles** [comment]
How effective against blob?

**Comment by SonixApache** [comment]
not very

File: code/game/machinery/ammolathe.dm [hunk 1]
```diff
@@ -73,6 +73,8 @@
 		new /obj/item/ammo_storage/magazine/a12mm/ops/empty(), \
 		new /obj/item/ammo_storage/magazine/a75/empty(), \
 		new /obj/item/ammo_storage/box/a75(), \
+		new /obj/item/ammo_casing/shotgun/dragonsbreath(), \
+		new /obj/item/weapon/storage/box/dragonsbreathshells, \
```
Comment (N3X15) [inline code/game/machinery/ammolathe.dm]: NO PARENS, YOU FUCKED UP THE FENG SHUI OF THIS LIST

File: code/modules/projectiles/ammunition/bullets.dm [hunk 2]
```diff
@@ -127,46 +127,55 @@
 	icon_state = "blshell"
 	projectile_type = /obj/item/projectile/bullet/blank
 	starting_materials = list(MAT_IRON = 250)
-	w_type = RECYK_METAL
```
Comment (N3X15) [inline code/modules/projectiles/ammunition/bullets.dm:130]: Why

File: code/modules/projectiles/ammunition/bullets.dm [hunk 3]
```diff
@@ -127,46 +127,55 @@
...
+	icon_state = "bdbshell"
+	projectile_type = /obj/item/projectile/bullet/fire_plume/dragonsbreath
+	starting_materials = list(MAT_IRON = 12500, MAT_PLASMA = 2000)
+	
+/obj/item/ammo_casing/shotgun/frag
+	name = "explosive shell"
+	desc = "A 12 gauge shell filled with a high-explosive mixture, for heavy anti-personnel usage."
+	icon_state = "fragshell"
+	projectile_type = /obj/item/projectile/bullet/boombullet
+	starting_materials = list(MAT_IRON = 12500, MAT_PLASMA = 8000)
```
Comment (N3X15) [inline code/modules/projectiles/ammunition/bullets.dm]: Seems a bit odd to me that the flammable thing would have less plasma than the explosive thing, but w/e.

Comment (SonixApache) [reply code/modules/projectiles/ammunition/bullets.dm]: Highly compressed plasma pellets cause explosions when they expand on impact.

File: code/modules/projectiles/ammunition/bullets.dm (hunk 2 above)
Comment (SonixApache) [reply code/modules/projectiles/ammunition/bullets.dm:130]: i have no idea

File: code/game/machinery/ammolathe.dm (hunk 1 above)
Comment (SonixApache) [reply code/game/machinery/ammolathe.dm]: fug

File: code/modules/projectiles/ammunition/bullets.dm (hunk 2 above)
Comment (SonixApache) [reply code/modules/projectiles/ammunition/bullets.dm:130]: because it's already defined in the parent class*

**Review by N3X15 — APPROVED** [review]

File: code/modules/projectiles/ammunition/bullets.dm [hunk 4]
```diff
@@ -127,46 +127,55 @@
...
 	icon_state = "bsshell"
 	projectile_type = /obj/item/projectile/bullet/buckshot
 	starting_materials = list(MAT_IRON = 12500)
+	
+/obj/item/ammo_casing/shotgun/dragonsbreath
+	name = "dragon's breath shell"
+	desc = "A 12 gauge shell filled with an incendiary mixture, for lighting up dark areas or setting things on fire."
+	icon_state = "bdbshell"
+	projectile_type = /obj/item/projectile/bullet/fire_plume/dragonsbreath
+	starting_materials = list(MAT_IRON = 12500, MAT_PLASMA = 2000)
```
Comment (MadmanMartian) [inline code/modules/projectiles/ammunition/bullets.dm]: If I'm reading this right, you get a 12250 iron return from recycling this when making it.

Seems all of the ammolathe stuff is inconsistent

**Comment by SonixApache** [comment]
🎉

**Pull request merged**